qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeActions.java

```diff
@@ -186,15 +201,101 @@ private static void doCodeActionToDisableValidation(Template template, List<Diag
 		codeActions.add(disableValidationForTemplateQuickFix);
 	}
 
+	/**
+	 * Create CodeAction for deprecated `with` Qute syntax.
+	 *
+	 * @param template    the Qute template.
+	 * @param diagnostic  the diagnostic list that this CodeAction will fix.
+	 * @param codeActions the list of CodeActions to perform.
+	 * @throws BadLocationException
+	 *
+	 */
+	private void doCodeActionsForNotRecommendedWithSection(Template template, Diagnostic diagnostic,
+			List<CodeAction> codeActions) {
+		Range withSectionRange = diagnostic.getRange();
+		try {
+			// 1. Retrieve the #with section node using diagnostic range
+			int withSectionStart = template.offsetAt(withSectionRange.getStart());
+			WithSection withSection = (WithSection) template.findNodeAt(withSectionStart + 1);
+
+			// 2. Initialize TextEdit array
+			List<TextEdit> edits = new ArrayList<TextEdit>();
+
+			// 2.1 Create text edit to update section start from #with to #let
+			// and collect all object parts from expression for text edit
+			// (e.g. {name} -> {#let name=item.name})
+			edits.add(createWithSectionOpenEdit(template, withSection));
+
+			// 2.2 Create text edit to update section end from /with to /let
+			edits.add(createWithSectionCloseEdit(template, withSection));
+
+			// 3. Create CodeAction
+			CodeAction replaceWithSection = CodeActionFactory.replace(QUTE_DEPRICATED_WITH_SECTION, edits,
+					template.getTextDocument(), diagnostic);
+			codeActions.add(replaceWithSection);
+		} catch (BadLocationException e) {
+			return;
+		}
+	}
+
+	private TextEdit createWithSectionOpenEdit(Template template, WithSection withSection) throws BadLocationException {
+		String objectPartName = withSection.getObjectParameter() != null ? withSection.getObjectParameter().getName()
+				: "";
+
+		// Use set to avoid duplicate parameters
+		HashSet<String> withExpressions = new HashSet<String>();
+
+		// Retrieve all expressions in #with section
+		findExpressions(withSection, withExpressions);

```

Comment:
Its not an expression that you collect, it's an object part. Please use objectPart (findObjectParts, )